openssl/src/aes.rs

```diff
@@ -1,15 +1,66 @@
-//! Low level AES functionality
+//! Low level AES IGE functionality
 //!
-//! The `symm` module should be used in preference to this module in most cases.
+//! AES ECB, CBC, XTS, CTR, CFB, GCM and other conventional symmetric encryption
+//! modes are found in [`symm`].  This is the implementation of AES IGE.
+//!
+//! Advanced Encryption Standard (AES) provides symmetric key cipher that
+//! the same key is used to encrypt and decrypt data.  This implementation
+//! uses 128, 192, or 256 bit keys.  This module provides functions to
+//! create a new key with [`new_encrypt`] and perform an encryption/decryption
+//! using that key with [`aes_ige`].
+//!
+//! AES is a 128-bit (16 byte) block cipher.  The rust implmentation will panic
```

Comment:
There are 128 and 256 bit variants of AES.

I'm not sure what the panic note is referring to - aes_ige doesn't have any asserts around the lengths of the input and output buffers.

Reply:
@sfackler I think it is refering to those lines
``` rust
/// Performs AES IGE encryption or decryption
///
/// # Panics
///
/// Panics if `in_` is not the same length as `out`, if that length is not a multiple of 16, or if
/// `iv` is not at least 32 bytes.
pub fn aes_ige(in_: &[u8], out: &mut [u8], key: &AesKey, iv: &mut [u8], mode: Mode) {
    unsafe {
        assert!(in_.len() == out.len());
        assert!(in_.len() % ffi::AES_BLOCK_SIZE as usize == 0);
        assert!(iv.len() >= ffi::AES_BLOCK_SIZE as usize * 2);
```

Doesn't it panic if the buffer lengths is not multiple of ffi::AES_BLOCK_SIZE ?

Reply:
I can see the confusion.  I'll try to incorporate both of those details at the top (the key is 128-bit, 192-bit, or 256-bit and that the input must be bounded to 16-bytes)